app/controllers/SalesFromEuListController.scala

```diff
@@ -48,25 +49,46 @@ class SalesFromEuListController @Inject()(
           val canAddCountries = number < Country.euCountries.size
           val list            = SalesFromEuSummary.addToListRows(request.userAnswers, mode)
 
-          Ok(view(form, mode, list, period, canAddCountries))
+          withCompleteData[SalesFromEuWithOptionalVat](
+            data = getIncompleteFromEuSales _,
+            onFailure = (incompleteSales: Seq[SalesFromEuWithOptionalVat]) => {
+            Ok(view(form, mode, list, period, canAddCountries, incompleteSales.map(_.countryOfSale)))
+          })(Ok(view(form, mode, list, period, canAddCountries)))
       }
   }
 
-  def onSubmit(mode: Mode, period: Period): Action[AnyContent] = cc.authAndGetData(period) {
+  def onSubmit(mode: Mode, period: Period, incompletePromptShown: Boolean): Action[AnyContent] = cc.authAndGetData(period) {
     implicit request =>
-      getNumberOfSalesFromEu {
-        number =>
+      withCompleteData[SalesFromEuWithOptionalVat](
+        data = getIncompleteFromEuSales _,
+        onFailure = (_: Seq[SalesFromEuWithOptionalVat]) => {
+        if(incompletePromptShown) {
+          firstIndexedIncompleteSaleFromEu match {
+            case Some(incompleteSalesFromCountry) if incompleteSalesFromCountry._1.salesFromCountry.isEmpty =>
+              Redirect(routes.CountryOfConsumptionFromEuController.onPageLoad( mode,  period, Index(incompleteSalesFromCountry._2), Index(0)))
```

Comment:
Will it always be index(0)?

Reply:
Yes, this is for when a country of sale has been specified, but no countries of consumption have been specified, therefore country of consumption index will always be 0